sklearn/metrics/tests/test_common.py

```diff
@@ -257,6 +257,7 @@
 
     "macro_f0.5_score", "macro_f1_score", "macro_f2_score",
     "macro_precision_score", "macro_recall_score",
+    "roc_auc_score",
```

Comment:
I don't think this affects any common tests. There are two uses of `METRICS_WITH_POS_LABEL`. One does not apply, I think; the other I suspect should apply but is being explicitly limited to a handful of metrics in a way that is very ugly and IMO inappropriate...

(Although I'm also confused why `roc_auc_score` is in `METRICS_UNDEFINED_BINARY` rather than `METRICS_UNDEFINED_MULTICLASS`.)

I'd appreciate if you could do a bit of an audit of the common tests wrt `roc_auc_score` and related metrics.